src/neon/neon_butterflies.rs

```diff
@@ -82,6 +82,36 @@ macro_rules! boilerplate_fft_neon_f32_butterfly {
                 }
```

Comment:
Out of place version should forward to immutable version.

Reply:
Actually I have a different recommendation. There's an incisistency between this and the SSE implementation, and imo the SSE implementation does it a better way. In that one, in the f32 and f64 macros there is no distinction between immutable and out of place versions, it's just the same method call. And the common implementations on line 194 and 213 just both call the same `perform_oop_fft_butterfly_multi` method